test/gtest/uct/test_md.cc

```diff
@@ -917,6 +917,79 @@ UCS_TEST_SKIP_COND_P(test_md, exported_mkey,
     ASSERT_UCS_OK(status);
 }
 
+UCS_TEST_P(test_md, rkey_compare_params_check)
+{
+    uct_rkey_compare_params_t params = {};
+    ucs_status_t status;
+    int result;
+
+    status = uct_rkey_compare(GetParam().component, 0, 0, &params, NULL);
+    ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
+
+    params.field_mask = UCS_BIT(0);
+    status = uct_rkey_compare(GetParam().component, 0, 0, &params, &result);
+    ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
+}
+
+// SM case is covered by XPMEM which has registration capability
+UCS_TEST_SKIP_COND_P(test_md, rkey_compare, !check_caps(UCT_MD_FLAG_REG))
+{
+    size_t size                      = 4096;
+    void *address                    = NULL;
+    uct_rkey_compare_params_t params = {};
+    std::vector<uint8_t> rkey_buffer1(md_attr().rkey_packed_size + 1);
+    std::vector<uint8_t> rkey_buffer2(md_attr().rkey_packed_size + 1);
+    uct_rkey_bundle_t b1, b2;
+    ucs_status_t status;
+    uct_mem_h memh1, memh2;
+    int result1, result2;
+
+    if (GetParam().md_name == "cuda_ipc") {
+        UCS_TEST_SKIP_R("missing IPC capability for registration");
+    }
+
+    status = ucs_mmap_alloc(&size, &address, 0, "test_rkey_compare_equal");
+    ASSERT_UCS_OK(status);
+    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh1);
+    ASSERT_UCS_OK(status);
+    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh2);
+    ASSERT_UCS_OK(status);
+    status = uct_md_mkey_pack(md(), memh1, &rkey_buffer1[0]);
+    ASSERT_UCS_OK(status);
+    status = uct_md_mkey_pack(md(), memh2, &rkey_buffer2[0]);
+    ASSERT_UCS_OK(status);
```

Comment:
minor:
this would look shorter
```suggestion
     ASSERT_UCS_OK(ucs_mmap_alloc(&size, &address, 0, "test_rkey_compare_equal");
     ASSERT_UCS_OK(reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh1));
     etc
```

Reply:
fixed